### PR TITLE
Bump toml_edit version

### PR DESCRIPTION
**Pull request by maciektr**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -49,7 +49,8 @@ snapbox = { version = "0.4.3", features = ["cmd", "path"] }
 tempfile = "3.3.0"
 test-case = "2.2.2"
 thiserror = "1.0.37"
-toml_edit = { version = "0.17.1", features = ["easy", "serde"] }
+toml = "0.7.1"
+toml_edit = { version = "0.18.1", features = ["easy", "serde"] }
```
Comment (mkaput) [inline Cargo.toml:53]: I see that there's `toml_edit 0.19.2` available. Could we bump to the latest one? https://github.com/toml-rs/toml/blob/main/crates/toml_edit/CHANGELOG.md

Comment (mkaput) [inline Cargo.toml:53]: The `easy` feature is deprecated: https://github.com/toml-rs/toml/blob/main/crates/toml_edit/CHANGELOG.md#0180---2023-01-23. We should remove usages of it

File: scarb/tests/e2e/build.rs [hunk 2]
```diff
@@ -225,8 +229,12 @@ fn compile_with_invalid_version() {
 error: failed to parse manifest at `[..]/Scarb.toml`
```
Comment (mkaput) [inline scarb/tests/e2e/build.rs:229]: Please use `indoc!` macro and its family, it will deal with newlines and indentation.

**Review by mkaput — CHANGES_REQUESTED** [review]

File: scarb/tests/e2e/build.rs [hunk 3]
```diff
@@ -193,14 +193,16 @@ fn compile_with_invalid_empty_name() {
...
+        .stdout_matches(indoc! {"\
+            error: failed to parse manifest at `[..]/Scarb.toml`
+
+            Caused by:
+                TOML parse error at line 3, column 20
+                  |
+                3 |             name = \"\"
+                  |                    ^^
+                empty string cannot be used as package name
+                \n",});
```
Comment (mkaput) [inline scarb/tests/e2e/build.rs:205]:
```suggestion
        .stdout_matches(indoc! {r#"
            error: failed to parse manifest at `[..]/Scarb.toml`

            Caused by:
                TOML parse error at line 3, column 20
                  |
                3 |             name = ""
                  |                    ^^
                empty string cannot be used as package name
        "#});
```

Apply this everywhere here. See how other tests do this: https://github.com/software-mansion/scarb/blob/dc08364119a5d19a132cddd566d9025d12e206d6/scarb/tests/e2e/build.rs#L65-L79

**Review by mkaput — CHANGES_REQUESTED** [review]

**Review by mkaput — APPROVED** [review]

File: scarb/tests/e2e/build.rs (hunk 3 above)
Comment (maciektr) [reply scarb/tests/e2e/build.rs:205]: Resolving after discussing offline.

**Pull request merged**